pkg/monitoring/feed_monitor.go

```diff
@@ -66,70 +68,98 @@ func (f *feedMonitor) Start(ctx context.Context) {
 		case <-ctx.Done():
 			return
 		}
-		// Map the payload.
-		var mapping map[string]interface{}
 		var err error
 		switch typed := update.(type) {
 		case StateEnvelope:
-			mapping, err = MakeConfigSetMapping(typed, f.solanaConfig, f.feedConfig)
-		case TransmissionEnvelope:
-			mapping, err = MakeTransmissionMapping(typed, f.solanaConfig, f.feedConfig)
-		default:
-			err = fmt.Errorf("unknown update type %T", update)
-		}
-		if err != nil {
-			f.log.Errorw("failed to map update", "error", err)
-			continue
-		}
-		// Encode the payload
-		var value []byte
-		switch update.(type) {
-		case StateEnvelope:
-			value, err = f.stateSchema.Encode(mapping)
-		case TransmissionEnvelope:
-			value, err = f.transmissionSchema.Encode(mapping)
-		default:
-			err = fmt.Errorf("unknown update type %T", update)
-		}
-		if err != nil {
-			f.log.Errorw("failed to encode to Avro", "payload", mapping, "error", err)
-			continue
-		}
-		// Push to kafka
-		var key = f.feedConfig.StateAccount.Bytes()
-		if err = f.producer.Produce(key, value); err != nil {
-			f.log.Errorw("failed to publish to kafka", "message", mapping, "error", err)
-			continue
-		}
-		// Publish metrics to prometheus
-		switch typed := update.(type) {
+			err = f.processState(typed)
+			if err != nil {
+				break
+			}
+			err = f.processConfigSetSimplified(typed)
```

Comment:
Just a not here, in this code, if `processState()` fails, then you don't do `processConfigSetSimplified()` or export to prometheus. This shouldn't be the case, each of these three pieces of functionality should be attempted in parallel. 

Let's do this in a later update!